lib/secret_handshake.dart

```diff
@@ -0,0 +1,50 @@
+class SecretHandshake {
```

Comment:
I don't know Dart very well but a couple of improvements that I could see here is using a bit mask to get the byte and shift instead of doing a division. Next, use a map between the secret move and the byte. Then, I think you could do it all in one loop instead of two.